libs/ledger-live-common/src/hw/extractOnboardingState.ts

```diff
@@ -0,0 +1,96 @@
+import { DeviceExtractOnboardingStateError } from "@ledgerhq/errors";
+const onboardingFlagsBytesLength = 4;
+
+const onboardedMask = 0x04;
+const inRecoveryModeMask = 0x01;
+const seedPhraseTypeMask = 0x60;
+const seedPhraseTypeFlagOffset = 5;
+const currentSeedWordIndexMask = 0x1f;
+
+export enum SeedPhraseType {
+  Twelve = "12-words",
+  Eighteen = "18-words",
+  TwentyFour = "24-words",
+}
+
+const fromBitsToSeedPhraseType = new Map<number, SeedPhraseType>([
+  [0, SeedPhraseType.TwentyFour],
+  [1, SeedPhraseType.Eighteen],
+  [2, SeedPhraseType.Twelve],
+]);
+
+export enum OnboardingStep {
+  WelcomeScreen = "WELCOME_SCREEN",
+  SetupChoice = "SETUP_CHOICE",
+  Pin = "PIN",
+  NewDevice = "NEW_DEVICE", // path "new device" & currentSeedWordIndex available
+  NewDeviceConfirming = "NEW_DEVICE_CONFIRMING", // path "new device" & currentSeedWordIndex available
+  RestoreSeed = "RESTORE_SEED", // path "restore seed" & currentSeedWordIndex available
+  SafetyWarning = "SAFETY WARNING",
+  Ready = "READY",
+}
+
+const fromBitsToOnboardingStep = new Map<number, OnboardingStep>([
+  [0, OnboardingStep.WelcomeScreen],
+  [1, OnboardingStep.SetupChoice],
+  [2, OnboardingStep.Pin],
+  [3, OnboardingStep.NewDevice],
+  [4, OnboardingStep.NewDeviceConfirming],
+  [5, OnboardingStep.RestoreSeed],
+  [6, OnboardingStep.SafetyWarning],
+  [7, OnboardingStep.Ready],
+]);
+
+export type OnboardingState = {
+  // Device not yet onboarded otherwise
+  isOnboarded: boolean;
+  // In normal mode otherwise
+  isInRecoveryMode: boolean;
+
+  seedPhraseType: SeedPhraseType;
+
+  currentOnboardingStep: OnboardingStep;
+  currentSeedWordIndex: number;
+};
+
+export const extractOnboardingState = (flagsBytes: Buffer): OnboardingState => {
+  if (!flagsBytes || flagsBytes.length < onboardingFlagsBytesLength) {
+    throw new DeviceExtractOnboardingStateError(
+      "Incorrect onboarding flags bytes"
+    );
+  }
+
+  const isOnboarded = !!(flagsBytes[0] & onboardedMask);
```

Comment:
you can also use Boolean(flagsBytes[0] & onboardedMask) here, I think. Will make it more explicit.